atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -0,0 +1,263 @@
+        :param pdc_secret_path: path to pdc.token file; $SOURCE_SECRET_PATH otherwise
+        :param pdc_contact_role: name of PDC role to contact
+        :param smtp_url: URL of SMTP server to use to send the message (e.g. "foo.com:25")
+        :param from_address: the "From" of the notification email
+        :param error_addresses: list of email addresses where to send an email if there's an error
+            (e.g. if we can't find out who to notify about the failed build)
+        """
+        super(SendMailPlugin, self).__init__(tasker, workflow)
+        self.url = url
+        self.send_on = send_on
+        self.pdc_url = pdc_url
+        self.pdc_verify_cert = pdc_verify_cert
+        self.pdc_component_df_label = pdc_component_df_label
+        self.pdc_secret_path = pdc_secret_path
+        self.pdc_contact_role = pdc_contact_role
+        self.smtp_url = smtp_url
+        self.from_address = from_address
+        self.error_addresses = error_addresses
+
+    def _should_send(self, rebuild, success, canceled):
+        """Return True if mail should be sent under given conditions and `self.send_on`."""
+        should_send = False
+
+        should_send_mapping = {
+            self.MANUAL_SUCCESS: not rebuild and success,
+            self.MANUAL_FAIL: not rebuild and not success,
+            self.AUTO_SUCCESS: rebuild and success,
+            self.AUTO_FAIL: rebuild and not success,
+            self.AUTO_CANCELED: rebuild and canceled
+        }
+
+        for state in self.send_on:
+            should_send |= should_send_mapping[state]
+        return should_send
+
+    def _render_mail(self, rebuild, success, canceled):
+        """Render and return subject and body of the mail to send."""
+        subject_template = 'Image %(image)s; Status %(endstate)s; Submitted by %(user)s'
+        body_template = '\n'.join([
+            'Image: %(image)s',
+            'Status: %(endstate)s',
+            'Submitted by: %(user)s',
+            'Logs: %(logs)s',
+        ])
+
+        endstate = None
+        if canceled:
+            endstate = 'canceled'
+        else:
+            endstate = 'successful' if success else 'failed'
+        url = None
+        if self.url and self.workflow.openshift_build_selflink:
+            url = urljoin(self.url, self.workflow.openshift_build_selflink + '/log')
+
+        formatting_dict = {
+            'image': self.workflow.image,
+            'endstate': endstate,
+            'user': '<autorebuild>' if rebuild else 'TODO user',
+            'logs': url
+        }
+        return (subject_template % formatting_dict, body_template % formatting_dict)
+
+    def _get_pdc_token(self):
+        if self.pdc_secret_path is not None:
+            token_file = os.path.join(self.pdc_secret_path, self.PDC_TOKEN_FILE)
+        else:
+            token_file = os.path.join(os.environ['SOURCE_SECRET_PATH'], self.PDC_TOKEN_FILE)
```

Comment:
The `sourceSecret` slot is already taken by pulp, so it's `pdc_secret_path` or nothing.